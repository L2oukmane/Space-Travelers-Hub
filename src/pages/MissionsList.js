import React from 'react';
import { useSelector } from 'react-redux';
import Mission from '../components/Mission';
import styles from './MissionsList.module.css';

const MissionsList = () => {
  const missions = useSelector((state) => state.missions);
  return (
    <ul className={styles.missionContainer}>
      <li className={styles.missionHeader}>
        <h3>Mission</h3>
        <h3>Description</h3>
        <h3>Status</h3>
      </li>
      {
        missions.length > 0
          ? missions.map((mission) => (
            <Mission
              key={mission.id}
              mission={mission}
            />
          ))
          : ''
      }
    </ul>
  );
};

export default MissionsList;
