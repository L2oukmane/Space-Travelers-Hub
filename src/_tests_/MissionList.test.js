import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import MissionsList from '../pages/MissionsList';
import testStore from '../redux/configureStore';

const mission = [
  {
    desc: 'Eutelsat S.A. is a European satellite operator',
    key: 'special',
    id: 1,
    name: 'defines the satellite',
    reservation: true,
  },
];

describe('testing the rendering of missions', () => {
  it('', () => {
    const { container } = render(
      <Provider store={testStore}>
        <MissionsList key={mission[0]} mission={mission[0]} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});