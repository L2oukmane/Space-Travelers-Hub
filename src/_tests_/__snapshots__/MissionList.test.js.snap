// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of missions  1`] = `
<div>
  <ul
    class="missionContainer"
  >
    <li
      class="missionHeader"
    >
      <h3>
        Mission
      </h3>
      <h3>
        Description
      </h3>
      <h3>
        Status
      </h3>
    </li>
  </ul>
</div>
`;